### PR TITLE
Impl lazy commitment for rust zk trie

### PR DESCRIPTION
**Pull request by i-m-aditya**
Closing #22

### CODE REVIEW

**Comment by lispc** [comment]
Add @noel2004 and @omerfirmak as reviewer. Not high priority pr. I know you are busy these days.

**Review by noel2004 — APPROVED** [review]
I think there is no issue in correction. Just some notes for possible optimization about cost.

**Review by lispc — APPROVED** [review]
(don't need to do extra micro optimizations now, especially at the cost of readablity)

File: rs_zktrie/src/raw.rs [hunk 1]
```diff
@@ -93,6 +101,9 @@ impl<H: Hashable, DB: ZktrieDatabase, const MAX_LEVELS: usize> ZkTrieImpl<H, DB,
             writable: true,
             root_hash: root,
             debug: false,
+            lock: RwLock::new(()),
```
[the remark this answers is not in the document]
Comment (lightsing) [reply rs_zktrie/src/raw.rs:104]: I observed a cycle count increase if only execute one block.

**Pull request merged**